plane/src/proxy/connection_monitor.rs

```diff
@@ -28,26 +34,29 @@ impl ConnectionMonitor {
 
     pub fn touch_backend(&mut self, backend_id: &BackendName) {
         match self.backends.entry(backend_id.clone()) {
-            Entry::Occupied(entry) => {
-                entry.get().1.store(true, Ordering::Relaxed);
+            Entry::Occupied(mut entry) => {
+                entry.get_mut().had_recent_connection = true;
             }
             Entry::Vacant(entry) => {
                 if let Some(listener) = &self.listener {
                     listener(backend_id);
                 }
                 self.visit_queue
                     .push_back((SystemTime::now() + HEARTBEAT_INTERVAL, backend_id.clone()));
-                entry.insert((AtomicU32::new(0), AtomicBool::new(false)));
+                entry.insert(BackendEntry {
+                    active_connections: 0,
+                    had_recent_connection: true,
+                });
             }
         }
     }
 
     pub fn inc_connection(&mut self, backend_id: &BackendName) {
         match self.backends.entry(backend_id.clone()) {
-            Entry::Occupied(entry) => {
-                let (active_connections, had_recent_connection) = entry.get();
-                active_connections.fetch_add(1, Ordering::SeqCst);
-                had_recent_connection.store(true, Ordering::SeqCst);
+            Entry::Occupied(mut entry) => {
+                let backend_entry = entry.get_mut();
+                backend_entry.active_connections += 1;
+                backend_entry.had_recent_connection = true;
```

Comment:
Why aren't the atomics required? Is the proxy not multithreaded? Can't different threads overwrite each other here?

Reply:
The `ConnectionMonitor` is [already in a `Mutex`](https://github.com/drifting-in-space/plane/blob/c1e771c87af44cbb52b3d9357b6a500bd810f527/plane/src/proxy/connection_monitor.rs#L80), so the atomics are redundant.

Reply:
Ah gotcha!